.github/actions/init_rust_job/action.yml

```diff
@@ -0,0 +1,77 @@
+name: CLI rust job init
+description: Base steps for CLI jobs
+
+inputs:
+  cdn:
+    type: choice
+    description: CDN to use when fetching assets
+    required: true
+    options:
+      - assets.grafbase.dev
+      - assets.grafbase.com
+  platform:
+    type: choice
+    description: Target platform to use when setting up rust
+    required: true
+    options:
+      - windows
+      - macos
+      - linux
+  assets_version:
+    description: Version of the assets to fetch from CDN
+    required: true
+
+runs:
+  using: 'composite'
+  steps:
+    - name: move CLI to root
+      shell: bash
+      run: |
+        shopt -s extglob
+        rm -r !("cli")
+        mv cli/* .
+        rm -r cli
+
+    - name: Cargo cache
+      uses: actions/cache@v3
+      continue-on-error: false
+      with:
+        key: ${{ runner.os }}-${{ runner.arch }}-cargo-${{ hashFiles('Cargo.lock') }}
+        restore-keys: ${{ runner.os }}-${{ runner.arch }}-cargo-
+        path: |
+          ~/.cargo/
+          target/
+
+    - name: Install rustup linux
+      if: ${{ inputs.platform == 'linux' }}
+      shell: bash
+      working-directory: cli
+      run: |
+        curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- -y --default-toolchain=stable --profile=minimal
+
+    - name: Install rustup windows
```

Comment:
IIRC windows also needs to add the VSCode build tools to the path

Reply:
Why is it needed?

Reply:
yeah the build works, hopefully the .exe works as well :)
did you get a change to test it? I don't have a windows at hand :P

Reply:
Not at the moment but the tests should have run it on Windows.
As long as there is an .exe it should be fine, the issue was that it needs the tools to be in the path to build